docker/deployment/app-deployment.yaml

```diff
@@ -57,10 +70,19 @@ spec:
       labels:
         app: app
     spec:
+      securityContext: # see https://github.com/sillsdev/web-languageforge/pull/974 for further explanation of the need for a securityContext here
+        fsGroup: 33 # this is www-data from /etc/passwd, k8s requires explicit assignment otherwise the volume will get mounted with root and www-data won't be allowed to create folders on-the-fly in there.
```

Comment:
I think we should also use `fsGroupChangePolicy: "OnRootMismatch"` (details [here](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#configure-volume-permission-and-ownership-change-policy-for-pods)) so that the `chown` operation will only happen once. The `assets` folder in Language Forge typically gets LOTS of files uploaded into it, because most people who use the audio feature upload one file per entry, so a single project could have over five thousand audio files. Multiply that by the number of projects, and that's a LOT of `chown` operations to run recursively every time the volume is mounted. But by using `fsGroupChangePolicy: "OnRootMismatch"` then we won't have to run the operation every time.

Note that the Kubernetes docs say this is available starting in Kubernetes version 1.20. Our Rancher server in Dallas is running v1.20.4 (verified by running `kubectl version`), and any commercial Kubernetes service we may choose to use will almost certainly have at least v1.20, so I believe this will work.

Reply:
this really helpful, I agree with you, it's probably a good approach to take anytime we have to do this...I wish it were the default actually.  I'll commit a change for this before merging